api/s2n.h

```diff
@@ -2291,7 +2291,22 @@ S2N_API extern int s2n_shutdown_send(struct s2n_connection *conn, s2n_blocked_st
 /**
  * Used to declare what type of client certificate authentication to use.
  *
- * Currently the default for s2n-tls is for neither the server side or the client side to use Client (aka Mutual) authentication.
+ * A s2n_connection will enforce client certificate authentication (mTLS) differently based on
+ * the `s2n_cert_auth_type` and `s2n_mode` (client/server) of the connection, as described below.
+ *
+ * Server behavior:
+ * - None(default): Will not request client authentication.
+ * - Optional: Request the client's certificate and validate it. If no certificate is received then
+ *      no validation is performed.
+ * - Required: Request the client's certificate and validate it. Abort the handshake if a client
+ *   certificate not recieved.
+ *
+ * Client behavior:
+ * - None: Abort the handshake if the server requests client authentication.
+ * - Optional(default): Sends the application provided client certificate if the server
+ *      requests client authentication.
```

Comment:
Importantly, the client can also send an empty Certificate message if the application doesn't provide one, which was the main reason the default behavior was changed in https://github.com/aws/s2n-tls/pull/4390. Maybe we can include something about this, so it's clear the application isn't forced to add a cert?
```suggestion
 * - Optional(default): Sends a client certificate when the server requests client
 *   authentication, if a certificate was provided by the application.
```

Reply:
My [previous version](https://github.com/aws/s2n-tls/pull/4454/commits/bdef9e1e0052383b7e741f21f1f623ff87dd7eb2) of this docs mentioned "empty" certs but after speaking with @maddeleine we thought best to avoid discussion of TLS messages in the docs and simply .

Once you start mentioning empty certs, you might want to start mentioning them in other sections. Also you need to start talking about TLS messages and explain the difference between Certificate message and the client cert within that message ("the client sends a Certificate message, which can contain an empty client cert").

Reply:
However, you did mention [PR](https://github.com/aws/s2n-tls/pull/4390). Do you think its worth trying to explain the empty cert behavior?

Reply:
Spoke with Sam and thinking we say this instead:

> - Optional(default): Sends the client certificate if the server requests client authentication. No certificate is sent if the application hasn't provided a certificate.